test/javascript/setupTests.js

```diff
@@ -0,0 +1,15 @@
+import '@testing-library/jest-dom/extend-expect'
+import { waitFor } from '@testing-library/react'
+import { queryCache } from 'react-query'
+
+afterEach(async () => {
+  queryCache.cancelQueries()
+  queryCache.clear()
+})
+
+afterEach(async () => {
+  // waitFor is important here. If there are queries that are being fetched at
+  // the end of the test and we continue on to the next test before waiting for
+  // them to finalize, the tests can impact each other in strange ways.
```

Comment:
Ouch! Great catch. Was there are reason for this code to be in a separate `afterEach` block? I sort of feel like it belongs to the code in the other `afterEach` block.

Reply:
Yes. You're correct! I've noticed that some of the tests fail because of this. In the other PRs I've opened, I've combined them into one block.